prime-router/src/main/kotlin/transport/RESTTransport.kt

```diff
@@ -91,48 +93,55 @@ class RESTTransport(private val httpClient: HttpClient? = null) : ITransport {
+                            restTransportInfo,
+                            reportId,
+                            jksCredential,
+                            credential,
+                            logger
+                        )
+                        logger.info("Token successfully added!")
 
-                    // post the report
-                    val reportClient = httpClient ?: createDefaultHttpClient(jksCredential, bearerTokens)
-                    val response = postReport(
-                        reportContent,
-                        fileName,
-                        restTransportInfo.reportUrl,
-                        httpHeaders,
-                        logger,
-                        reportClient
-                    )
-                    val responseBody = response.bodyAsText()
-                    // update the action history
-                    val msg = "Success: REST transport of $fileName to $restTransportInfo:\n$responseBody"
-                    logger.info("Message successfully sent!")
-                    actionHistory.trackActionResult(response.status, msg)
-                    actionHistory.trackSentReport(
-                        receiver,
-                        sentReportId,
-                        fileName,
-                        restTransportInfo.toString(),
-                        msg,
-                        header.reportFile.itemCount
-                    )
-                    actionHistory.trackItemLineages(Report.createItemLineagesFromDb(header, sentReportId))
+                        // post the report
+                        val reportClient = httpClient ?: createDefaultHttpClient(jksCredential, bearerTokens)
+                        val response = postReport(
+                            reportContent,
+                            fileName,
+                            restTransportInfo.reportUrl,
+                            httpHeaders,
+                            logger,
+                            reportClient
+                        )
+                        val responseBody = response.bodyAsText()
+                        // update the action history
+                        val msg = "Success: REST transport of $fileName to $restTransportInfo:\n$responseBody"
+                        logger.info("Message successfully sent!")
+                        actionHistory.trackActionResult(response.status, msg)
+                        actionHistory.trackSentReport(
+                            receiver,
+                            sentReportId,
+                            fileName,
+                            restTransportInfo.toString(),
+                            msg,
+                            header.reportFile.itemCount
+                        )
+                        actionHistory.trackItemLineages(Report.createItemLineagesFromDb(header, sentReportId))
+                    } catch (t: Throwable) {
+                        throw t
```

Comment:
@halprin, yes I do need the try/catch.  If I don't have it there, it won't throw exception when there is an exception throw in the block of code.  Since, it is running in different thread/context.